src/components/ChangePlan/PlanCalculator.js

```diff
@@ -186,74 +215,120 @@ const PlanCalculator = ({
+                                {getPlanFee(planData.plan)}
+                              </span>
+                            </span>
+                          ) : (
+                            <></>
+                          )}
+                          <h2 className="dp-price-large">
+                            <span className="dp-price-large-money">US$</span>
+                            <span className="dp-price-large-amount">
+                              {planData.discount?.discountPercentage
+                                ? Math.round(
+                                    getPlanFee(planData.plan) *
+                                      (1 - planData.discount?.discountPercentage / 100),
+                                  )
+                                : getPlanFee(planData.plan)}
+                            </span>
+                          </h2>
+                          <span className="dp-for-time">Por mes</span>
+                          <div class="dp-agreement">
+                            {planData.discount?.discountPercentage ? (
+                              <p>
+                                El pago anual será de
+                                <strong>
+                                  {' '}
+                                  US$
+                                  {Math.round(
+                                    getPlanFee(planData.plan) *
+                                      (1 - planData.discount.discountPercentage / 100) *
+                                      planData.discount.monthsAmmount,
+                                  )}
+                                </strong>
+                              </p>
+                            ) : (
+                              <></>
+                            )}
+                            <p>
+                              La renovación es automática y puedes cancelarla cuando quieras. Estos
+                              valores no incluyen impuestos.
+                            </p>
+                          </div>
+                        </div>
+                      </div>
+                    </div>
+                  </div>
+                </div>
+              </article>
+            </section>
+            <div className="dp-container">
+              <div className="dp-rowflex">
+                <div className="dp-align-center dp-cta-plans">
+                  <Link
+                    className="dp-button button-medium primary-grey"
+                    to={`/plan-selection${safePromoId ? `?promo-code=${safePromoId}` : ''}`}
+                  >
+                    Volver a Planes
+                  </Link>
+
+                  <a
+                    className="dp-button button-medium primary-green"
+                    href={
+                      _('common.control_panel_section_url') +
+                      `/AccountPreferences/UpgradeAccountStep2?IdUserTypePlan=${planData.plan.id}&fromStep1=True` +
```

Comment:
Perhaps this link could be generated elsewhere like in the plan service for example. What do you think?

Reply:
Like this 
```
planService.getBuyLink(planData.plan.id, planData.discount?.id, safePromoId);
```